### PR TITLE
Error message instead of crash on invalid audio.newSource call

### PR DESCRIPTION
**Pull request by mcclure**
Checks for file not found and file-not-understood cases.

### CODE REVIEW

File: src/api/audio.c [hunk 1]
```diff
@@ -68,8 +68,14 @@ int l_lovrAudioNewSource(lua_State* L) {
     stream = *type;
   } else {
     Blob* blob = luax_readblob(L, 1, "Source");
+    if (!blob) // readblob responsible for setting Lua error
+      return 0;
```
Comment (bjornbytes) [inline src/api/audio.c]: I don't see how this is ever true?  Looking at readblob, it should either return a Blob or throw an error.

Comment (mcclure) [reply src/api/audio.c]: So I guess the correct thing to check is if (!blob->data) ?

File: src/api/audio.c [hunk 2]
```diff
@@ -68,8 +68,14 @@ int l_lovrAudioNewSource(lua_State* L) {
     stream = *type;
   } else {
     Blob* blob = luax_readblob(L, 1, "Source");
+    if (!blob->data) // readblob (1) never returns NULL (2) is responsible for setting Lua error
+      return 0;
```
Comment (bjornbytes) [inline src/api/audio.c]: I'm still confused: given `readblob` either returns a Blob or throws an error, we shouldn't need to check the return value and could just omit this `if`.

Comment (mcclure) [reply src/api/audio.c]: You are right, I messed up in my initial testing and thought this was necessary.

**Pull request merged**